hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -297,10 +306,58 @@ def _assert_eq(style, a, b):
     return f"assert {a} == {b}"
 
 
+def _imports_for_object(obj):
+    """Return the imports for `obj`, which may be empty for e.g. lambdas"""
+    try:
+        if (not callable(obj)) or obj.__name__ == "<lambda>":
+            return set()
+        name = _get_qualname(obj).split(".")[0]
+        return {(_get_module(obj), name)}
+    except Exception:
```

Comment:
I wonder whether it will be better to catch `AttributeError` and `RuntimeError` explicitly (as they are coming from `_get_module`). So other error types (if there are any) will not be hidden behind this `except Exception`. What do you think?

Reply:
It's also possible to get really weird runtime errors from e.g. descriptors which run arbitrary code when accessing the `.__name__` attribute!

Our error handling here is "if we can't work out what to import, don't import anything", so `except Exception` felt appropriate.  It's an unusual style though, specific to the ghostwriter's ethos of producing decent but editable output.

Reply:
Oh, I missed that! :) Makes sense